### PR TITLE
fix: handle image manifest in hash manager

### PR DESCRIPTION
**Pull request by TobiTenno**
### What did you fix? 
image cache not updating, making builds take a long time

---

### Reproduction steps
1. Make image cache update through cache manager
2. handle saves through a single method

---

### Evidence/screenshot/link to line

N/A

### Considerations
- Does this contain a new dependency? **[No]**
- Does this introduce opinionated data formatting or manual data entry? **[No]**
- Does this pr include updated data files in a separate commit that can be reverted for a clean code-only pr? **[No]**
- Have I run the linter? **[Yes]**
- Is is a bug fix, feature request, or enhancement? **[Bug Fix]**

### CODE REVIEW

**Comment by TobiTenno** [comment]
@AyAyEm looks good

**Comment by AyAyEm** [comment]
Maybe update the description and we're good to go

File: build/hashManager.mjs [hunk 1]
```diff
@@ -1,32 +1,39 @@
...
 
-    const compareHashs = () =>
-      Object.entries(this.exportCache).every(([key, { hash }]) => hash === exportCache[key]?.hash);
+    return oldCacheKeys.length === cacheKeys.length && compareHashes();
+  }
+
+  hasChanged(key) {
+    const keyInBoth = key in this.exportCache && key in previousExportCache;
+    if (!keyInBoth) {
+      console.warn(`Could not find the key: ${key} in both the saved cache and the current one`);
```
Comment (AyAyEm) [inline build/hashManager.mjs:33]: Maybe a better message here?

Comment (TobiTenno) [reply build/hashManager.mjs:33]:
```suggestion
      console.warn(`Key not found... ${key} missing`);
```

**Review by AyAyEm — APPROVED** [review]

**Pull request merged**